WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationForm.swift

```diff
@@ -133,32 +136,60 @@ struct BlazeCampaignCreationForm: View {
                 isShowingDevicePicker = false
             }
         }
+        .onChange(of: viewModel.errorState) { newValue in
+            isShowingErrorAlert = newValue != .none
+        }
+        .alert(isPresented: $isShowingErrorAlert, content: {
+            Alert(title: Text(Localization.ErrorAlert.error),
+                  message: Text(viewModel.errorState.errorMessage),
+                  primaryButton: .default(Text(Localization.ErrorAlert.retry), action: {
+                Task {
+                    switch viewModel.errorState {
+                    case .none:
+                        return
+                    case .fetchingAISuggestions:
+                        await viewModel.loadAISuggestions()
+                    }
+                }
+            }),
+                  secondaryButton: .cancel())
+        })
+        .task {
+            await viewModel.loadAISuggestions()
+        }
+        .task {
+            await viewModel.downloadProductImage()
+        }
     }
 }
 
 private extension BlazeCampaignCreationForm {
     var adPreview: some View {
         VStack(spacing: Layout.contentPadding) {
             VStack(alignment: .leading, spacing: Layout.contentMargin) {
-                // TODO: use product image here
-                // Product image
-                Image(uiImage: .blazeIntroIllustration)
-                    .resizable()
-                    .aspectRatio(contentMode: .fill)
-                    .cornerRadius(Layout.cornerRadius)
-
-                // TODO: dynamic content
+                // Image
+                if viewModel.isDownloadingImage {
+                    remoteImage
+                } else {
+                    localImage
+                }
```

Comment:
We download the image in order to send the `UIImage` to the "Edit data" screen. While we wait for the image to be downloaded, we use `KFImage` and the product image URL to show the image.

Reply:
Do you think this is a duplicated effort? The local image works exactly like KFImage AFAIK:
- It uses KingFisher under the hood so it has caching.
- It can observe the downloading and display a placeholder when it's in progress.